proto/gasless/v1beta1/gasless.proto

```diff
@@ -0,0 +1,70 @@
+syntax = "proto3";
+package rollapp.gasless.v1beta1;
+
+import "gogoproto/gogo.proto";
+import "google/protobuf/timestamp.proto";
+
+option go_package = "github.com/dymensionxyz/dymension-rdk/x/gasless/types";
+option (gogoproto.goproto_getters_all) = false;
+
+
+// AddressType enumerates the available types of a address.
+enum AddressType {
+    option (gogoproto.goproto_enum_prefix) = false;
+  
+    // the 32 bytes length address type of ADR 028.
+    ADDRESS_TYPE_32_BYTES = 0 [(gogoproto.enumvalue_customname) = "AddressType32Bytes"];
+    // the default 20 bytes length address type.
+    ADDRESS_TYPE_20_BYTES = 1 [(gogoproto.enumvalue_customname) = "AddressType20Bytes"];
+  }
+
+message TxGTIDs {
+    string tx_path_or_contract_address = 1;
+    repeated uint64 gas_tank_ids = 2;
+}
+
+message GasTank {
+    uint64 id = 1;
+    string provider = 2;
+    string reserve = 3;
+    bool is_active = 4;
+    uint64 max_txs_count_per_consumer = 5;
+    string max_fee_usage_per_consumer = 6 [(gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int", (gogoproto.nullable) = false];
+    string max_fee_usage_per_tx = 7 [(gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int", (gogoproto.nullable) = false];
+    repeated string txs_allowed = 8;
+    repeated string contracts_allowed = 9;
+    repeated string authorized_actors = 10;
+    string fee_denom = 11;
```

Comment:
>Why not support multiple fee denoms per gas tank?
Can be done, but to keep the tings simple for going out only single denom is allowed by tanks.
Further, if there is need for the same the feature can be added in future and rolled out the upgrades.

>Because, when we try to check if a gas tank can be used as fee source, we pass its ID, but then we compare the fee denom, which can lead to an error
This is not true, `GetFeeSource` is the method responsible for getting the fee source initially, but within it it depend on `CanGasTankBeUsedAsSource`, the `GetFeeSource` pass on the gas tank ID to the `CanGasTankBeUsedAsSource` and it is the one which queries the gastank and make comparisons between fee denom. So there is no possibilites for the exceptions to be occurred